src/Voron/Data/Tables/TableSchema.IndexDefinitions.cs

```diff
@@ -348,6 +363,8 @@ public void Validate()
 
                 if (GenerateKey.Method.GetCustomAttribute<StorageIndexEntryKeyGeneratorAttribute>() == null)
                     throw new ArgumentOutOfRangeException(nameof(GenerateKey), $"{nameof(GenerateKey)} must be marked with custom attribute '{nameof(StorageIndexEntryKeyGeneratorAttribute)}'");
+
+                //  TODO : validate 'OnEntryChanged'
```

Comment:
Keep in mind that I already plan to use it for my own things.. like to keep track of the last change vector.. 
Let's assume I will add this functionally after the release, what needs to be done, to ensure proper schema upgrade?